dask_bigquery/core.py

```diff
@@ -53,6 +57,7 @@ def _stream_to_dfs(bqs_client, stream_name, schema, read_kwargs):
 def bigquery_read(
     make_create_read_session_request: callable,
     project_id: str,
+    credentials: Credentials,
```

Comment:
Ideally you'd allow a wider set of credentials than this. [google.auth.credentials.Credentials](https://googleapis.dev/python/google-auth/latest/reference/google.auth.credentials.html#google.auth.credentials.Credentials) is going to be the most flexible.

There are two types of credentials that come to mind that you'll want to support that aren't service account credentials:

* [External Account Credentials](https://googleapis.dev/python/google-auth/latest/reference/google.auth.external_account.html#google.auth.external_account.Credentials) -- These are useful for folks running on other clouds.
* [OAuth 2.0 Credentials](https://googleapis.dev/python/google-auth/latest/reference/google.oauth2.credentials.html#google.oauth2.credentials.Credentials) -- Also known as end-user credentials. These are useful for folks running locally. It's the "open a browser window and use your Google account" flow.